pkg/services/oracle/neofs/neofs.go

```diff
@@ -60,30 +57,34 @@ func Get(ctx context.Context, priv *keys.PrivateKey, u *url.URL, addr string, re
 	prmi.SetDefaultSigner(neofsecdsa.Signer(priv.PrivateKey))
 	c, err = client.New(prmi)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create client: %w", err)
+		return nil, nil, fmt.Errorf("failed to create client: %w", err)
 	}
 
 	var prmd client.PrmDial
 	prmd.SetServerURI(addr)
 	prmd.SetContext(ctx)
 	err = c.Dial(prmd) //nolint:contextcheck // contextcheck: Function `Dial->Balance->SendUnary->Init->setNeoFSAPIServer` should pass the context parameter
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	defer c.Close()
+	var (
+		finalizer = func() { c.Close() }
+		res       io.ReadCloser
+	)
 
 	switch {
 	case len(ps) == 0 || ps[0] == "": // Get request
-		return getPayload(ctx, c, objectAddr, resReader)
+		res, err = getPayload(ctx, c, objectAddr)
 	case ps[0] == rangeCmd:
-		return getRange(ctx, c, objectAddr, resReader, ps[1:]...)
+		res, err = getRange(ctx, c, objectAddr, ps[1:]...)
 	case ps[0] == headerCmd:
-		return getHeader(ctx, c, objectAddr)
+		res, err = getHeader(ctx, c, objectAddr)
 	case ps[0] == hashCmd:
-		return getHash(ctx, c, objectAddr, ps[1:]...)
+		res, err = getHash(ctx, c, objectAddr, ps[1:]...)
 	default:
-		return nil, ErrInvalidCommand
+		err = ErrInvalidCommand
 	}
+	return res, finalizer, err
```

Comment:
With smth like
```
if err != nil || resBytes != nil {
    c.Close()
    if err != nil {
        return nil, err
    }
    return io.NopCloser(bytes.NewReader(resBytes)), nil
}
return clientCloseWrapper(res, c), nil
}

type clientCloseWrapper struct {
    io.ReadCloser
    c *client.Client
}
func (w clientCloseWrapper) Close() error {
    res := w.ReadCloser.Close()
    w.c.Close()
    return res
}
```

Reply:
I also thought about it, but to be honest, the previous code (the one we had before SDK update) looks better for me in this situation.

Fixed.